dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/handlers/DimensionDataCloudControlErrorHandler.java

```diff
@@ -66,9 +66,9 @@ public void handleError(HttpCommand command, HttpResponse response) {
             exception = new AuthorizationException(message, exception);
             break;
          case 404:
-            if (!command.getCurrentRequest().getMethod().equals("DELETE")) {
-               exception = new ResourceNotFoundException(message, exception);
-            }
+            // CloudControl uses error code 400 with RESOURCE_NOT_FOUND to report missing assets
+            // 404 means malformed URI only
+            exception = new IllegalStateException(message, exception);
```

Comment:
A malformed URL in a request is better represented as an `IllegalArgumentException`. Illegal sates are more for 409 (Conflict) errors.

Reply:
thanks, done